pkg/services/control/server/list_objects.go

```diff
@@ -0,0 +1,45 @@
+package control
+
+import (
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/engine"
+	"github.com/nspcc-dev/neofs-node/pkg/services/control"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (s *Server) ListObjects(req *control.ListObjectsRequest, stream control.ControlService_ListObjectsServer) error {
+	// verify request
+	if err := s.isValidRequest(req); err != nil {
+		return status.Error(codes.PermissionDenied, err.Error())
+	}
+
+	// check availability
+	err := s.ready()
+	if err != nil {
+		return status.Error(codes.Internal, err.Error())
+	}
+
+	objectsIds, err := engine.List(s.storage, 0)
```

Comment:
huge number of objects can overload the server unnecessarily. Consider https://github.com/nspcc-dev/neofs-node/blob/1a9784c11ecc9f41193a4b99c5171ad470afb016/pkg/local_object_storage/engine/list.go#L55-L62

pay attention to `Does not include inhumed objects` statement

Reply:
Done with `ListWithCursor`. Set maximum amount of addresses from func to 1024, but not sure about this number.